src/elements/netspeak-graph-body.tsx

```diff
@@ -0,0 +1,502 @@
+
+									const highlighted =
+										classes.filter(val => props.highlightedPhrases.includes(val)).length > 0;
+
+									return (
+										<g
+											key={"c" + cIndex + "e" + eIndex + "g"}
+											className="textG"
+											fontSize={fontSize}>
+											<rect
+												// x={(width / props.columns.length) * (cIndex + 0.5)}
+												y={element.yPosition! - fontSize}
+												id={indexes + "rect"}
+												fill="none"
+												height={fontSize}
+												key={index + "rect"}></rect>
+
+											<text
+												onMouseEnter={() => onMouseEnterText(classes)}
+												onMouseLeave={() => onMouseLeaveText()}
+												// onMouseLeave={() => setIsShown(false)}>™
+												// x={(width / props.columns.length) * (cIndex + 0.5)}
+												onClick={() => props.toggleWordSelection(element, indexes)}
+												y={element.yPosition}
+												id={indexes + "text"}
+												key={indexes + eIndex}
+												className={
+													"graphHighlightable" +
+													(props.selectedWords.filter(word => word.id == indexes).length > 0
+														? " selectedWord"
+														: "") +
+													(selected ? " selectedPath" : "") +
+													(highlighted ? " highlightedPath" : "")
+												}>
+												{element.text}
+											</text>
+
+											{
+												// Create link paths
+												element.previous
+													?.filter((item, index) => {
+														return element.previous?.indexOf(item) == index;
+													})
+													.map(previousText => {
+														const sourceIndexes: { cIndex: number; eIndex: number }[] = [];
+														const previousElements = props.columns[
+															cIndex - 1
+														].elements.filter((previousElement, pEIndex) => {
+															if (previousText == previousElement.text) {
+																sourceIndexes.push({
+																	cIndex: cIndex - 1,
+																	eIndex: pEIndex,
+																});
+																return true;
+															}
+															return false;
+														});
+														//seed links
+														previousElements.map(linkedPrevElement => {
+															const sourceIndex = sourceIndexes.shift();
+															//set offset from other links based index of source element. if only 1 source element, offset by target element.
+
+															let offsetFactor: number;
+															if (props.columns[cIndex - 1].elements.length > 1) {
+																offsetFactor = props.columns[
+																	cIndex - 1
+																].elements.indexOf(linkedPrevElement);
+															} else {
+																offsetFactor =
+																	MAX_ROWS -
+																	props.columns[cIndex].elements.indexOf(element);
+															}
+															const linkedPhrases = linkedPrevElement.phrases.filter(
+																phraseIn => {
+																	return element.phrases
+																		.flatMap(a => {
+																			return a.text;
```

Comment:
This return is 18 (!!) indentation levels deep. Maybe we could extract from of the stuff of this >100 lines-long return expression out into functions?